libp2p/connection.nim

```diff
@@ -15,28 +15,33 @@ import peerinfo,
        varint,
        vbuffer
 
+logScope:
+  topic = "Connection"
+
 const DefaultReadSize* = 1 shl 20
 
 type
   Connection* = ref object of LPStream
     peerInfo*: PeerInfo
     stream*: LPStream
     observedAddrs*: Multiaddress
+    maxReadSize: int
 
   InvalidVarintException = object of LPStreamError
   InvalidVarintSizeException = object of LPStreamError
 
 proc newInvalidVarintException*(): ref InvalidVarintException =
-  newException(InvalidVarintException, "unable to parse varint")
+  newException(InvalidVarintException, "Unable to parse varint")
 
 proc newInvalidVarintSizeException*(): ref InvalidVarintSizeException =
-  newException(InvalidVarintSizeException, "wrong varint size")
+  newException(InvalidVarintSizeException, "Wrong varint size")
 
-proc init*[T: Connection](self: var T, stream: LPStream) =
+proc init*[T: Connection](self: var T, stream: LPStream, maxReadSize = DefaultReadSize) =
```

Comment:
seriously, it really doesn't make sense at all to have a default max read size here - it's not valid for _any_ of the protocols that implement `Connection`: signal is 8mb, noise is 64kb, mplex is 1mb _payload_ + dynamic header size (protobuf overhead not counted, so one needs to add a reasonable header overhead to max read size to be compatible) etc - we will never be following any spec if we use this read size

Reply:
But at each step, we create a new connection anyways, so for `secio` you can set the size to 8mb, for mplex it would be 1mb (per channel), so on and so forth. I don't see where this breaks?

Reply:
BTW, I can move this out so we don't clutter this PR, since this is about proper channel cleanup. I'll move this out, thats probably best.

Reply:
let's look at this the other way around: when is the default value useful? why not have it as a parameter that you have to supply when calling `init`?

it's a bit out of scope for this PR indeed

Reply:
I removed this changes and will create a new PR so we can continue discussing it there.